cmd/kubeadm/app/phases/apiconfig/setupmaster.go

```diff
@@ -49,6 +50,8 @@ func findMyself(client *clientset.Clientset) (*v1.Node, error) {
 }
 
 func attemptToUpdateMasterRoleLabelsAndTaints(client *clientset.Clientset) error {
+	validate.WaitForNode(client)
```

Comment:
I understand why we need to wait for the master node to register here. My question is: why are we also waiting for it to be ready? That won't happen until a network daemonset is applied. This seems like it would cause the same deadlock as https://github.com/kubernetes/kubernetes/issues/43815.